wsd/FileServer.cpp

```diff
@@ -786,7 +786,13 @@ void FileServerRequestHandler::readDirToHash(const std::string &basePath, const
                 strm.next_out = (unsigned char *)&cbuf[0];
                 strm.total_out = strm.total_in = 0;
 
-                deflate(&strm, Z_FINISH);
+                const int deflateResult = deflate(&strm, Z_FINISH);
+                if (deflateResult != Z_OK && deflateResult != Z_STREAM_END)
+                {
+                    LOG_ERR("Failed to deflate, result: " << deflateResult);
+                    free(cbuf);
+                    break;
```

Comment:
It looks a bit inconsistent that here we go for "break" after logging the error, but above pre-existing code goes with "continue" after logging the error. I assume these typically don't fail in practice; so if they do, it would be better to also do "break" above? But that's pre-existing code. Just sharing my concerns. :-)

Reply:
Yeah the outer continue on deflateInit2 failure matches the existing stat failure continue. Unsure if there is a great solution on failures of these.